pkg/apis/pipeline/v1beta1/param_types.go

```diff
@@ -236,23 +237,28 @@ func ArrayReference(a string) string {
 	return strings.TrimSuffix(strings.TrimPrefix(a, "$("+ParamsPrefix+"."), "[*])")
 }
 
-func validatePipelineParametersVariablesInTaskParameters(params []Param, prefix string, paramNames sets.String, arrayParamNames sets.String) (errs *apis.FieldError) {
+func validatePipelineParametersVariablesInTaskParameters(params []Param, prefix string, paramNames sets.String, arrayParamNames sets.String, objectParamNameKeys map[string][]string) (errs *apis.FieldError) {
```

Comment:
Would you mind adding a docstring for this function and the one below it?

Reply:
Done!

Reply:
thanks! the docstring should ideally focus on what the function does, not which other functions call it, so I'd remove the reference to PipelineTask (both here and in other docstrings)

Reply:
Changed! Thanks for the suggestion!